File: Greatest_Number/greatest_number.cpp
#include <iostream>

//Function returns the index of the greatest number in the array

int greatest(int* v, int i, int k){

    if(i == k)
        return i;
    else{
        //Dividir
        int j = (i+k)/2;

        //Conquer
        int l = greatest(v, i, j);
        int r = greatest(v, i+1, k);

        //Combine
        if(v[l] > v[r])
            return l;
        else
            return r;
    }
}

int main(){

    int n = 0;
    std::cout << "Size of array: ";
    std::cin >> n;
    int vector[n];

    std::cout << "Enter elements: ";
    for(int i = 0; i < n; i++)
        std::cin >> vector[i];

    int gr = greatest(vector, 0, n-1);
    std::cout << "The index of the greatest number is: " << gr;
}

File: Greatest_Number/greatest_number.py
#Function returns the index of the greatest number in the array

def greatest(v, i, k):
    if(i == k):
        return i
    else:
        #Divide
        j = int((i+k)/2)

        #Conquer
        l = greatest(v, i, j)
        r = greatest(v, i+1, k)

        #Combine
        if(v[l] > v[r]):
            return l
        else:
            return r

vector = []
n = int(input("Size of array: "))

for i in range(n):
    num = int(input("Enter elements: "))
    vector.append(num)

gr = greatest(vector, 0, n-1)
print("The index of the greatest number is:", gr)
